Sources/TABResourceLoader/Protocols/Resource types/NetworkJSONDictionaryResourceType.swift

```diff
@@ -29,10 +29,4 @@
 import Foundation
 
 /// Defines a resource that can be fetched from a network where the root type is a JSON object
-public protocol NetworkJSONDictionaryResourceType: NetworkResourceType, JSONDictionaryResourceType {}
-
-public extension NetworkJSONDictionaryResourceType {
-  var httpHeaderFields: [String: String]? {
-    return ["Content-Type": "application/json"]
-  }
-}
+public protocol NetworkJSONDictionaryResourceType: NetworkJSONResourceType, JSONDictionaryResourceType {}
```

Comment:
i'm not so sure about this, i feel like we need a bit of a beat up around it. it's not clear from the name that the request body needs to be JSON (if you need to send data) and also that the response type is JSON. seems like this was an existing issue in TRL and why we needed this PR in the first place. maybe we need to look at removing some of these types and leave it to the consumer to compose the resource types they want (that would obviously be a breaking change 😢 ). i also think it might be nice to make a clearer distinction between the resource request information (`NetworkJSONResourceType`) and the resource response information (`JSONDictionaryResourceType`). i find the namings all a bit confusing at the moment... but i could just be out of touch as i didn't build this to start with. @Neil3079 @lucianomarisi any thoughts?

as a side note i think the change in `NetworkResourceType.swift` is spot on and where we should be heading for

Reply:
i was thinking and commenting exactly the same @deanbrindley87. see my comments 😄